### ISSUE TITLE
Allow multiple mirror-servers to be used simultanioulsly

### ISSUE THREAD

**Issue by TorfDrottel** [linuxmint/mintupdate#597]
To speed-up the updates, it would be nice to distribute the downloads of packages to a bunch of selected mirror-servers.

The user should be able to select those, probably by filtering by country and bandwidth.

If only one Server is selected, this would be the current behaviour. Otherwise, the fastest not currently occupied server will get the next download. If any of those take to long, an alternative server can be selected for a package to bypass the currently busy server.

If seen this behaviour in manjaro and found the update process amazing fast.

In case of an recent update, only those mirrors can be used that already have the most recent package. This can reduce the potential download speed, and means additional code complexity, but also allow users to get the update as soon as any of the mirrors got the update.

**Comment by ronyAhmed1200** [comment]
Yeah this is a important feature to have.. Oh, i noticed this request was from 2019 two year ago. But linux mint devs didn't even look at it... 

I guess that's why ppl are leaving mint and switching to another distro..
